examples/prover/src/main.rs

```diff
@@ -71,15 +108,48 @@ fn prover_example() {
         task_manager.add_task(task.clone());
     }
 
-    let root_receipt = task_manager.run();
+    let conditional_receipt = task_manager.run();
+
+    let output = conditional_receipt
+        .claim
+        .as_value()
+        .unwrap()
+        .output
+        .as_value()
+        .unwrap()
+        .as_ref()
+        .unwrap();
+    let assumptions = output.assumptions.as_value().unwrap();
+
+    let coprocessor = RefCell::borrow(&coprocessor);
+    let mut succinct_receipt = conditional_receipt.clone();
+    for assumption in assumptions.iter() {
+        let assumption = assumption.as_value().unwrap();
+        println!("{assumption:?}");
+        let assumption_receipt = coprocessor.receipts.get(&assumption.claim).unwrap().clone();
+        let opts = ProverOpts::default();
+        succinct_receipt = client
+            .resolve(
+                &opts,
+                succinct_receipt.try_into().unwrap(),
+                assumption_receipt.try_into().unwrap(),
+                AssetRequest::Inline,
+            )
+            .unwrap();
```

Comment:
This emulates the way Bonsai works. I think @mothran might have a better sense of whether this will go into the work plan but last we talked it sounds like it was going to be part of TaskDB, not necessarily the work plan.